File: 0049-group-anagrams/solution.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& s) {

        unordered_map < string  , vector < string >> mp ;
        for(auto i : s ){
         string t(i);
            sort(t.begin(),t.end());
            mp[t].push_back(i);
        }
        
        vector < vector <string >> v ;
        for(auto i : mp){
            v.push_back( i.second) ;
        }
        
        return v ; 
    }
};

File: 0049-group-anagrams/solution.java
class Solution {
    public static List<List<String>> groupAnagrams(String[] strs) {
        int n = strs.length;
        Map<String,List<String>> mp = new HashMap<>();
        List<List<String>> res = new ArrayList<>();
        for(int i = 0 ; i < n ;i++){
            char[] c = strs[i].toCharArray();
            Arrays.sort(c);
            String s = new String(c);
            if(mp.containsKey(s)){
                List<String> list = mp.get(s);
                list.add(strs[i]);
                mp.put(s,list);
            }
            else {
                List<String> list = new ArrayList<>();
                list.add(strs[i]);
                mp.put(s,list);
            }
        }

        for(Map.Entry<String,List<String>> e : mp.entrySet()){
            res.add(e.getValue());
        }
        return res;
    }
}

// aet - eat , tea , ate
// ant - tan , nat ,
// abt - bat
